File: src/main/java/br/com/blz/testjava/domain/model/Product.java
package br.com.blz.testjava.domain.model;

import java.io.Serializable;
import java.util.Objects;

public class Product implements Serializable {

  private Long sku;
  private String name;
  private Inventory inventory;

  public Long getSku() {
    return sku;
  }

  public void setSku(Long sku) {
    this.sku = sku;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Inventory getInventory() {
    return inventory;
  }

  public void setInventory(Inventory inventory) {
    this.inventory = inventory;
  }

  public Boolean isMarketable() {
    return (inventory == null ? false : inventory.getQuantity() > 0);
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Product product = (Product) o;
    return Objects.equals(sku, product.sku) &&
      Objects.equals(name, product.name) &&
      Objects.equals(inventory, product.inventory);
  }

  @Override
  public int hashCode() {
    return Objects.hash(sku, name, inventory);
  }
}

File: src/main/kotlin/br/com/blz/testjava/domain/model/Product.kt
package br.com.blz.testjava.domain.model

import com.fasterxml.jackson.annotation.JsonProperty
import javax.persistence.*

@Entity
open class Product (

    @Id
    open var sku: Long,

    @Column(nullable = false)
    open var name: String,

    @OneToOne(cascade = [CascadeType.ALL])
    open var inventory: Inventory

) {

    /**
     * <p>Verifica se o produto e comercializavel.</p>
     *
     * @return {@code true} se o produto for comercializavel
     *
     * @see Inventory#quantity
     */
    @JsonProperty("isMarketable")
    open fun isMarketable() = inventory != null && inventory.calculateTotalQuantity() > 0

}